### PR TITLE
@mixes support

### PR DESCRIPTION
**Pull request by justinfagnani**
- [ ] CHANGELOG.md has been updated

### CODE REVIEW

File: src/model/element.ts [hunk 1]
```diff
@@ -44,6 +49,20 @@ export class Element extends ElementBase implements Feature {
...
   superClass?: Reference;
+
+  /**
+   * Mixins that this class declares with `@mixes`.
+   *
+   * Mixins are applied linearly after the superclass, in order from first
+   * to last. Mixins that compose other mixins will be flattened into a
+   * single list. A mixin can be applied more than once, each time its
+   * members override those before it in the prototype chain.
+   */
```
Comment (rictic) [inline src/model/element.ts]: Nice docs!

File: src/polymer/polymer-element.ts [hunk 2]
```diff
@@ -384,6 +316,132 @@ function inheritFrom(element: PolymerElement, superElement: PolymerElement) {
...
+        sourceRange: scannedMixinReference.sourceRange!,
+      });
+      continue;
+    } else if (mixins.size > 1) {
+      element.warnings.push({
+        message:
+            `@mixes reference, multiple mixins found ${scannedMixinReference
+                .identifier}`,
+        severity: Severity.ERROR,
+        code: 'mixes-reference-not-found',
```
Comment (rictic) [inline src/polymer/polymer-element.ts]: s/not-found/-multiple/ in the code

File: src/polymer/polymer-element.ts [hunk 3]
```diff
@@ -384,6 +316,132 @@ function inheritFrom(element: PolymerElement, superElement: PolymerElement) {
...
+        sourceRange: scannedMixinReference.sourceRange!,
+      });
+      continue;
+    }
+    const mixin = mixins.values().next().value;
+    if (!(mixin instanceof PolymerElementMixin)) {
+      element.warnings.push({
+        message: `@mixes reference to a non-Mixin ${scannedMixinReference
```
Comment (rictic) [inline src/polymer/polymer-element.ts]: Add a second sentence with suggested fixes. e.g.

is it imported? does it have a `@polymerMixin` annotation?

File: src/polymer/polymer2-element-scanner.ts [hunk 4]
```diff
@@ -111,13 +111,33 @@ class ElementVisitor implements Visitor {
...
+                       .map((annotation) => {
+                         const mixinId = annotation.name;
+                         // TODO(justinfagnani): we need source ranges for jsdoc
+                         // annotations
+                         const sourceRange =
+                             this._document.sourceRangeForNode(node)!;
+                         if (mixinId == null) {
+                           warnings.push({
+                             code: 'class-mixes-annotation-no-id',
+                             message: '@mixes annotation with no identifier',
```
Comment (rictic) [inline src/polymer/polymer2-element-scanner.ts]: Give an example of correct syntax?

File: src/polymer/polymer-element.ts (hunk 2 above)
Comment (justinfagnani) [inline src/polymer/polymer-element.ts]: done

File: src/model/element.ts (hunk 1 above)
Comment (justinfagnani) [inline src/model/element.ts]: thanks!

File: src/polymer/polymer-element.ts (hunk 3 above)
Comment (justinfagnani) [inline src/polymer/polymer-element.ts]: done

File: src/polymer/polymer2-element-scanner.ts (hunk 4 above)
Comment (justinfagnani) [inline src/polymer/polymer2-element-scanner.ts]: done

**Pull request merged**